Tests/Runner/RunnerTest.php

```diff
@@ -1736,20 +1734,23 @@ public function testTokenObfuscate()
 
     public function testExecutorStoreUsage()
     {
+        $dataDir = '';
+        $usage = [];
         $this->clearConfigurations();
-        $job = new Job($this->getEncryptorFactory()->getEncryptor());
-        $jobMapperStub = self::getMockBuilder(JobMapper::class)
-            ->disableOriginalConstructor()
-            ->setMethods(['get', 'update'])
+        $usageFile = self::getMockBuilder(NullUsageFile::class)
```

Comment:
Nešlo by místo takovýhodle sofistikovanýho mocku vytvořit novou třídu, která přesně tohle bude dělat? místo callbacků by se to přesunulo do classy? přijde mi, že je tak v testech nepokrytej a neotestovanej kód, stejnej použitej na dvou místech.

Reply:
no tak z toho mocku radosti pochopitelne neskacu, ale je to tim, ze to ma takovej debilni interface - metoda `storeUsage` to cte primo z datovyho adresare a je timpadem stavova a tak ji bez toho stavu blbe otestuju, ze se skutecne zavolala

primarne jde v tomhle testu o to `$usageFile->expects(self::once())` ale chtel jsem tam mit i kontrolu toho vnitrniho stavu, aby se overilo, ze se to zavolalo ve spravny okamzik

myslis vytvorit tu tridu jen pro testy? nebo co by jinak mela delat? a jakej je tam nepokrytej a neotesovanej kod? vlastni usagefile tu neni, ten je tudle https://github.com/keboola/docker-runner-jobs/blob/odin-init/Tests/UsageFileTest.php a https://github.com/keboola/docker-runner-jobs/blob/odin-init/src/Docker/Runner/UsageFile/UsageFile.php

Reply:
Představoval jsem si to nějak takhle

```php
class TestUsageFile implements UsageFileInterface
{
    $private $dataDir;

    public function setDataDir($dataDir)
    {
        $this->dataDir = $dataDir;
    }

    public function storeUsage()
    {
        // intentionally blank
    }

    public function getUsageData()
    {
        $usageFileName = $this->dataDir . '/out/usage.json';
        return json_decode(file_get_contents($usageFileName), true);

    }
}
```
a k tomu 
```php
$usageFile = self::getMockBuilder(TestUsageFile::class)
    ->getMock();
$usageFile->expects(self::once())
    ->method('storeUsage');

$usage = $usageFile->getUsageData();
```
případně i nějaký další vyhodnocování přesunout `TestUsageFile` (po vzoru https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/TestHandler.php), ale to asi není tady vůbec potřeba

Reply:
> myslis vytvorit tu tridu jen pro testy?

stacilo napsat "ano" :D

a co myslis teda tim neotestovanym kodem?

Reply:
v podstatě jen toto
```php
$usageFileName = $dataDir . '/out/usage.json';
$usage[] = json_decode(file_get_contents($usageFileName), true);
```

Reply:
done